### PR TITLE
feat: allow instrumentation via base driver

### PR DESCRIPTION
**Pull request by Dharin-shah**
Instrumentation Object Schema:
 ```
{
"eventName" : "initializeDriver",
"duration" : 12220
"options" : {},
"children" : [{
                        "eventName" : "driverServerStart",
                        "duration": 8760,
                        "options" : {},
                        "children": []
                     }]
}
```

This data will help when building custom spans using opentelemetry plugin as well as works as a meta field for detailed instrumentation (which can be kept optional)

### CODE REVIEW

**Review by jlipps — COMMENTED** [review]
the code itself looks good but i'm not sure what the overall purpose of the _instrumentation object is so I'm finding it hard to do a complete review

**Comment by KazuCocoa** [comment]
Test code like `_eventHistory` for instrumentation also may help to understand the usage.

**Comment by Dharin-shah** [comment]
@KazuCocoa Yes will add the required tests. Also @jlipps updated description for more info. Will add the tests for usage clarity

File: lib/basedriver/driver.js [hunk 1]
```diff
@@ -88,9 +87,16 @@ class BaseDriver extends Protocol {
     // allow subclasses to have internal drivers
     this.managedDrivers = [];
 
-    // store event timings
-    this._eventHistory = {
-      commands: [] // commands get a special place
+    this._meta = {
+      _eventHistory: {
```
Comment (jlipps) [inline lib/basedriver/driver.js:91]: given that `_meta` is already "private", its subfields don't really need to be (i.e., we can get rid of `_` here I think)

File: lib/basedriver/driver.js [hunk 2]
```diff
@@ -160,15 +166,96 @@ class BaseDriver extends Protocol {
...
+      this.instrumentEventStart(eventName);
+      let result;
+      try {
+        result = fn.call(...args);
+        this.instrumentEventEnd(eventName);
+      } catch (error) {
+        this.instrumentEventEnd(eventName);
+        throw error;
+      }
+      return result;
```
Comment (jlipps) [inline lib/basedriver/driver.js:193]: if we return in the try, we don't need the separate `let` instantiation.

Comment (Dharin-shah) [reply lib/basedriver/driver.js]: done

File: lib/basedriver/driver.js (hunk 1 above)
Comment (Dharin-shah) [reply lib/basedriver/driver.js]: done

**Comment by mykola-mokhnach** [comment]
Closing this PR as we are not going to maintain this module anymore. It has been moved to `@appium` monorepo

**Pull request closed without merging**